pygac/gac_reader.py

```diff
@@ -256,8 +259,72 @@ def get_calibrated_channels(self):
                 self.scans["scan_line_number"],
                 chan,
                 self.spacecraft_name)
+
+        # Mask out corrupt values
+        mask = self.get_corrupt_mask()
+        channels[mask] = np.nan
+
+        # Apply KLM/POD specific postprocessing
+        self.postproc(channels)
+
+        # Mask pixels affected by scan motor issue
+        if self.is_tsm_affected():
+            LOG.info('Correcting for temporary scan motor issue')
+            self.mask_tsm_pixels(channels)
+
         return channels
 
+    def get_lonlat(self):
```

Comment:
I can imagine that scientific GAC users would sometimes want the original data, not corrected for clock drift, and/or not interpolated. Could we have optional arguments to switch these off ?

Reply:
Sure!